webpack.config.js

```diff
@@ -1,34 +1,33 @@
+const path = require("path"),
+  webpack = require("webpack"),
+  HtmlWebpackPlugin = require("html-webpack-plugin");
+
 module.exports = {
-  entry: "./src/index.tsx",
+  entry: {
+    app: ["./src/app/App.tsx", "webpack-hot-middleware/client"],
+    vendor: ["react", "react-dom"]
+  },
   output: {
-    filename: "bundle.js",
-    path: __dirname + "/dist"
+    path: path.resolve(__dirname, "dist"),
+    filename: "js/[name].bundle.js"
   },
-
-  // Enable sourcemaps for debugging webpack's output.
   devtool: "source-map",
-
   resolve: {
-    // Add '.ts' and '.tsx' as resolvable extensions.
-    extensions: [".ts", ".tsx", ".js", ".json"]
+    extensions: [".js", ".jsx", ".json", ".ts", ".tsx"]
   },
-
   module: {
     rules: [
-      // All files with a '.ts' or '.tsx' extension will be handled by 'awesome-typescript-loader'.
-      { test: /\.tsx?$/, loader: "awesome-typescript-loader" },
-
-      // All output '.js' files will have any sourcemaps re-processed by 'source-map-loader'.
+      {
+        test: /\.(ts|tsx)$/,
+        loader: "ts-loader"
```

Comment:
Just out of curiosity, is there any specific reason to change the loader? If you want to change it, please remove awesome-typescript-loader from package.json as well.